PanoramaPublic/src/org/labkey/panoramapublic/PanoramaPublicModule.java

```diff
@@ -0,0 +1,169 @@
+import org.labkey.api.pipeline.PipelineService;
+import org.labkey.api.protein.ProteinService;
+import org.labkey.api.security.permissions.AdminPermission;
+import org.labkey.api.security.roles.RoleManager;
+import org.labkey.api.settings.AdminConsole;
+import org.labkey.api.targetedms.TargetedMSService;
+import org.labkey.api.view.ActionURL;
+import org.labkey.api.view.BaseWebPartFactory;
+import org.labkey.api.view.Portal;
+import org.labkey.api.view.ShortURLService;
+import org.labkey.api.view.ViewContext;
+import org.labkey.api.view.WebPartFactory;
+import org.labkey.api.view.WebPartView;
+import org.labkey.panoramapublic.pipeline.CopyExperimentPipelineProvider;
+import org.labkey.panoramapublic.proteomexchange.SubmissionDataValidator;
+import org.labkey.panoramapublic.security.CopyTargetedMSExperimentRole;
+import org.labkey.panoramapublic.view.expannotations.TargetedMSExperimentWebPart;
+import org.labkey.panoramapublic.view.expannotations.TargetedMSExperimentsWebPart;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class PanoramaPublicModule extends SpringModule
+{
+    public static final String NAME = "PanoramaPublic";
+
+    @Override
+    public String getName()
+    {
+        return NAME;
+    }
+
+    @Override
+    public double getVersion()
+    {
+        return 19.22;
+    }
+
+    @Override
+    public boolean hasScripts()
+    {
+        return true;
+    }
+
+    @Override
+    protected void init()
+    {
+        addController(PanoramaPublicController.NAME, PanoramaPublicController.class);
+        PanoramaPublicSchema.register(this);
+    }
+
+    @Override
+    protected void startupAfterSpringConfig(ModuleContext moduleContext)
+    {
+        PipelineService service = PipelineService.get();
+        service.registerPipelineProvider(new CopyExperimentPipelineProvider(this));
+
+        // Register the CopyExperimentRole
+        RoleManager.registerRole(new CopyTargetedMSExperimentRole());
+
+        // Add a link in the admin console to manage journals.
+        ActionURL url = new ActionURL(PanoramaPublicController.JournalGroupsAdminViewAction.class, ContainerManager.getRoot());
+        AdminConsole.addLink(AdminConsole.SettingsLinkType.Configuration, "panorama public", url, AdminPermission.class);
+
+        PanoramaPublicListener listener = new PanoramaPublicListener();
+        ExperimentService.get().addExperimentListener(listener);
+        ContainerManager.addContainerListener(listener);
+
+        ShortURLService shortUrlService = ShortURLService.get();
+        shortUrlService.addListener(listener);
+
+        TargetedMSService tmsService = TargetedMSService.get();
+        tmsService.registerSkylineDocumentImportListener(listener);
+        tmsService.registerTargetedMSFolderTypeListener(listener);
+
+        // TODO !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

Comment:
Need to make protein, peptide and modification searches work in the top level "Panorama Public" project.  This will require the "search in subfolders" checkbox to be hidden and value set to true by default.  The results returned should include a column for the experiment(s) that contains the matches.